src/complex/Core/Parameters/utils/FilePathGenerator.hpp

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include "complex/Core/Parameters/utils/StackFileListInfo.hpp"
+#include "complex/complex_export.hpp"
+
+#include <array>
+#include <cstdint>
+#include <filesystem>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <fmt/core.h>
+
+namespace fs = std::filesystem;
+
+namespace complex
+{
+
+namespace FilePathGenerator
+{
+
+std::vector<std::string> GenerateFileList2(const StackFileListInfo& info, bool validatePaths)
+{
+  std::vector<std::string> fileList;
+  // make sure the input path exists. If it does *not* then just bail out now.
+  if(validatePaths && !fs::exists(info.getInputPath()))
+  {
+    return fileList;
+  }
+  int32_t index = 0;
+  // Create the actual format string
+  std::stringstream fmt_string_stream;
+  fmt_string_stream << "{}/{}{:0" << info.getPaddingDigits() << "d}{}{}";
```

Comment:
```suggestion
  std::string format_string = fmt::format("{{}}/{{}}{{:0{}d}}{{}}{{}}", info.paddingDigits);
```
We can use fmt here instead to format the format string.